carta/structs.py

```diff
@@ -0,0 +1,16 @@
+import posixpath
+
+from .validation import validate, String, Constant
+from .constants import Polarization, PROTO_POLARIZATION
+
+
+class StokesImage:
```

Comment:
This class needs a docstring. Have a look at `util.Macro` to see how to write it. For the main description I'd put `An object which groups information about an image file to be used as a component in a Stokes hypercube.`

Reply:
@confluence, I added some documentation to the `StokesImage` class.
Nevertheless, when describing the `stokes` parameter, I quoted the `carta.constants.Polarization` object, but the stokes index of `carta.constants.Polarization` is not exactly the same with `PROTO_POLARZATION` and of course I can't specify the parameter like `carta.constants.PROTO_POLARIZATION`. What should I do to specify the correct index for `stokes` parameter?

Reply:
The user doesn't need to know anything about `PROTO_POLARIZATION`. The user passes in a `Polarization` enum as a parameter, *not* a number which matches the protobuf numbering. You are doing the mapping inside the function by extracting the index attribute from the enum object.

Reply:
Longer explanation: in fact, `PROTO_POLARIZATION` is *just* used internally to set those index attributes in the Enum constructor, and is never used again.

The reason that it's not in the constructor's internal scope is 1) that would be slightly inefficient -- it would be recreated every time the constructor was called, and 2) we or the user might want to reuse this mapping for something, so it's useful to expose it. The reason it's not in the Enum's class scope is that it would pollute the Enum's namespace, and then we would have to do something else to exclude the name of the dict from the Enum values.

That's why I ended up putting those two theme palette mappings in the module's scope, and suggested the same for this mapping.